### PR TITLE
use arrow 6.1.0

### PR DESCRIPTION
**Pull request by Jimexist**
# Which issue does this PR close?

Closes #.

 # Rationale for this change

# What changes are included in this PR?

# Are there any user-facing changes?

### CODE REVIEW

**Review by houqp — APPROVED** [review]

File: ballista-examples/Cargo.toml [hunk 1]
```diff
@@ -29,7 +29,7 @@ publish = false
 rust-version = "1.56"
 
 [dependencies]
-arrow-flight = { version = "6.0.0" }
+arrow-flight = { version = "6.1.0" }
```
Comment (alamb) [inline ballista-examples/Cargo.toml:32]: I think these versions need to be something like `^6.1.0` to force using arrow 6.1.0 or newer

Comment (houqp) [reply ballista-examples/Cargo.toml:32]: @Dandandan mentioned that in my previous PR that by default cargo treats `6.1.0` as `^6.1.0`: https://doc.rust-lang.org/cargo/reference/specifying-dependencies.html#specifying-dependencies-from-cratesio

Comment (Jimexist) [reply ballista-examples/Cargo.toml:32]: maybe we should use "6.1" instead of "6.1.0" to make it less restrictive.

Comment (houqp) [reply ballista-examples/Cargo.toml:32]: yeah, that's a good idea.

Comment (Dandandan) [reply ballista-examples/Cargo.toml:32]: FTR, They are all functionally equivalent.
`=` and `^` both mean semver compatible. Adding the final zero has no effect (but might be more clear to remove it)

Source:
https://doc.rust-lang.org/cargo/reference/specifying-dependencies.html#specifying-dependencies-from-cratesio

Comment (houqp) [reply ballista-examples/Cargo.toml:32]: ha, you are right. patch version only makes sense if we need it to be higher than a particular patch release, i.e. greater than 0.

**Pull request merged**